src/yaudio.cpp

```diff
@@ -71,13 +72,17 @@ static void set_note_defaults();
 static void start_i2s();
 
 ////////////////////////////// Public Functions ///////////////////////////////
-void add_notes(const std::string &new_notes) {
+bool add_notes(const std::string &new_notes) {
     // If WAVE is running, then stop it and reset the audio buffer.
     // Otherwise if notes are running, it's fine to let them keep running.
     if (wave_running) {
         stop();
     }
 
+    if ((notes.length() + new_notes.length()) > MAX_NOTES_IN_BUFFER) {
+        return false;
```

Comment:
@jgoeders, do you have thoughts on what we should do if the size is too big? Right now I just give up and return false. We could add the remaining notes to the buffer to fill it up but then the user wouldn't know what got put into the buffer and what didn't get put in the buffer.

Reply:
I don't think buffering the remaining notes is any different that just allowing a longer string.  I think we just need to throw them away.  The current issue is that the board will crash and run out of memory if they call this function in a loop.